R/add_thresholds_transition_risk.R

```diff
@@ -0,0 +1,130 @@
+#' Adds low and high thresholds for transition risk score
+#'
+#' @param emissions_profile_products A dataframe
+#' @param all_uuids_scenario_sectors A dataframe
+#' @param scenarios A dataframe
+#'
+#' @keywords internal
+#'
+#' @return A dataframe.
+#' @export
+#'
+#' @examples
+#' library(tiltToyData)
+#' library(readr)
+#' library(dplyr)
+#' options(readr.show_col_types = FALSE)
+#'
+#' emissions_profile_products <- read_csv(toy_emissions_profile_products_ecoinvent())
+#' all_uuids_scenario_sectors <- read_csv(toy_sector_profile_companies()) |>
+#'   select(-c("companies_id", "company_name", "clustered")) |>
+#'   distinct()
```

Comment:
@maurolepore A toy dataset can be added for `all_uuids_scenario_sectors` in `tiltToyData` with similar code. Should we add it or not?

Reply:
Yeah, it would be nice. For now you can create a `toy_*()` right here and move it to tiltToyData later. Best to create it in a separate file so it's easy to move.

``` r
# Created with
# readr::read_csv(tiltToyData::toy_sector_profile_companies()) |>
#   dplyr::select(-c("companies_id", "company_name", "clustered")) |>
#   dplyr::distinct() |> 
#   datapasta::tribble_paste()

toy_all_uuids_scenario_sectors <- function() {
  file <- tempfile(fileext = ".csv")
  
  tibble::tribble(
  # styler: on
               ~activity_uuid_product_uuid, ~isic_4digit,   ~tilt_sector,            ~tilt_subsector, ~type,     ~sector,       ~subsector,
    "76269c17-78d6-420b-991a-aa38c51b45b7",     "'4100'", "construction", "construction residential", "ipr", "buildings",               NA,
    "76269c17-78d6-420b-991a-aa38c51b45b7",     "'4100'", "construction", "construction residential", "weo",     "total",    "residential",
    "833caa78-30df-4374-900f-7f88ab44075b",     "'2591'",       "metals",             "other metals", "ipr",  "industry", "other industry",
    "833caa78-30df-4374-900f-7f88ab44075b",     "'2591'",       "metals",             "other metals", "weo",     "total",       "industry",
    "bf94b5a7-b7a2-46d1-bb95-84bc560b12fb",     "'2410'",       "metals",             "iron & steel", "ipr",  "industry", "iron and steel",
    "bf94b5a7-b7a2-46d1-bb95-84bc560b12fb",     "'2410'",       "metals",             "iron & steel", "weo",     "total", "iron and steel"
  # styler: off
    ) |> 
    write.csv(file, row.names = FALSE)
  
  file
}

toy_all_uuids_scenario_sectors()
#> [1] "/tmp/RtmpsbAnCp/file1170543905314.csv"

# Compare
tiltToyData::toy_emissions_profile_products_ecoinvent()
#> [1] "/usr/local/lib/R/site-library/tiltToyData/extdata/emissions_profile_products_ecoinvent.csv.gz"

readr::read_csv(toy_all_uuids_scenario_sectors(), show_col_types = FALSE)
#> # A tibble: 6 × 7
#>   activity_uuid_product_uuid isic_4digit tilt_sector tilt_subsector type  sector
#>   <chr>                      <chr>       <chr>       <chr>          <chr> <chr> 
#> 1 76269c17-78d6-420b-991a-a… '4100'      constructi… construction … ipr   build…
#> 2 76269c17-78d6-420b-991a-a… '4100'      constructi… construction … weo   total 
#> 3 833caa78-30df-4374-900f-7… '2591'      metals      other metals   ipr   indus…
#> 4 833caa78-30df-4374-900f-7… '2591'      metals      other metals   weo   total 
#> 5 bf94b5a7-b7a2-46d1-bb95-8… '2410'      metals      iron & steel   ipr   indus…
#> 6 bf94b5a7-b7a2-46d1-bb95-8… '2410'      metals      iron & steel   weo   total 
#> # ℹ 1 more variable: subsector <chr>
```

<sup>Created on 2024-06-05 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
Thanks for writing the code! :)